app/javascript/components/patient/vaccines/VaccineModal.js

```diff
@@ -0,0 +1,232 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Modal, Row, Col, Form } from 'react-bootstrap';
+import Select from 'react-select';
+import moment from 'moment';
+import ReactTooltip from 'react-tooltip';
+
+import DateInput from '../../util/DateInput';
+
+class VaccineModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product_name: this.props.currentVaccineData?.product_name,
+      // If there is not already a group name, default to the first option
+      group_name: this.props.currentVaccineData?.group_name
+        ? this.props.currentVaccineData.group_name
+        : this.props.group_name_options
+        ? this.props.group_name_options[0]
+        : null,
+      administration_date: this.props.currentVaccineData?.administration_date,
+      dose_number: this.props.currentVaccineData?.dose_number,
+      notes: this.props.currentVaccineData?.notes,
+      // Sorting so that the blank option shows up at the top
+      sorted_dose_number_options: this.props.dose_number_options ? this.props.dose_number_options.sort() : [],
+    };
+  }
+
+  /**
+   */
+  checkIfValid = () => {};
```

Comment:
I think this can be removed

Reply:
Good catch, removed here: https://github.com/SaraAlert/SaraAlert/pull/721/commits/66e9a6d482854054a97a2f0ac8c1ec4c0c6af3ca#diff-182e13a82e02db14eb5222d1eb289b14639dc17acd927608ff7e8b032cfb9af8L29